website/public/js/controllers/settingsCtrl.js

```diff
@@ -66,12 +66,23 @@ habitrpg.controller('SettingsCtrl',
       User.set({'flags.newStuff':true});
     }
 
-    $scope.saveDayStart = function(){
-      var dayStart = +User.user.preferences.dayStart;
-      if (_.isNaN(dayStart) || dayStart < 0 || dayStart > 24) {
+    $scope.passDayStart = User.user.preferences.dayStart;
+
+    $scope.saveDayStart = function(newDayStart){
+      var oldDayStart = User.user.preferences.dayStart;
+      var dayStart = newDayStart;
+      var lastCron = User.user.lastCron;
+      var getOldStart = Shared.startOfDay({ dayStart: oldDayStart});
+      var getNewStart = Shared.startOfDay({ dayStart: dayStart});
+      var isoNewStart = Shared.isoTimestamp(getNewStart);
+
+      if (dayStart == undefined || _.isNaN(dayStart) || dayStart < 0 || dayStart > 24) {
```

Comment:
Not sure if this was a copy paste typo or not but, `if ( || dayStart < 0 || dayStart > 24 ) {message}` is missing `!dayStart`. It should be: `if (!dayStart || dayStart < 0 || dayStart > 24 ) {message}`